### PR TITLE
Added flag to disable weight decay decoupling and new optimizers

### PR DESCRIPTION
**Pull request by BloodAxe**
* Add `decouple_wd` to OptimizerCallback to disable weight decay decoupling (needed for next two optimizers, that does decoupling internally)
* Lamb
* QHAdamW

### CODE REVIEW

File: catalyst/contrib/optimizers/lamb.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import collections
+
+import torch
+from torch.optim.optimizer import Optimizer
+from torch.utils.tensorboard import SummaryWriter
+
+
+def log_lamb_rs(optimizer: Optimizer, event_writer: SummaryWriter,
+                token_count: int):
```
Comment (Scitator) [inline catalyst/contrib/optimizers/lamb.py]: codestyle, please:
```
def log_lamb_rs(
    optimizer: Optimizer,
    event_writer: SummaryWriter,
    token_count: int
):
```

File: catalyst/contrib/optimizers/lamb.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                if i in state:
+                    results[i].append(state[i])
+
+    for k, v in results.items():
+        event_writer.add_histogram(f'lamb/{k}', torch.tensor(v), token_count)
+
+
+class Lamb(Optimizer):
+    def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-6,
+                 weight_decay=0, adam=False):
```
Comment (Scitator) [inline catalyst/contrib/optimizers/lamb.py]: codestyle

File: catalyst/contrib/optimizers/qhadamw.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import torch
+from torch.optim.optimizer import Optimizer
+
+
+class QHAdamW(Optimizer):
+    def __init__(self, params, lr=1e-3, betas=(0.995, 0.999),
+                 nus=(0.7, 1.0), weight_decay=0.0, eps=1e-8):
```
Comment (Scitator) [inline catalyst/contrib/optimizers/qhadamw.py]: codestyle

File: catalyst/dl/callbacks/optimizer.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ def __init__(
         accumulation_steps: int = 1,
         optimizer_key: str = None,
         loss_key: str = "loss",
+        decouple_wd = True
```
Comment (Scitator) [inline catalyst/dl/callbacks/optimizer.py]: let's use full naming, like `decouple_weight_decay` – it will be much clearer

Comment (Scitator) [reply catalyst/dl/callbacks/optimizer.py]: or even `weight_decay_decoupling`?

Comment (BloodAxe) [reply catalyst/dl/callbacks/optimizer.py]: Rename to `decouple_weight_decay`, i like the idea of having verb coming first.
